packages/react/src/input.tsx

```diff
@@ -0,0 +1,96 @@
+import * as React from "@hpcc-js/preact-shim";
+
+export interface IInput {
+    type: "number" | "range" | "date" | "text" | "select" | "checkbox";
+    value: number | string | boolean;
+    className?: string;
+    attributes?: any;
+    callbacks?: any;
+}
+
+export const Input: React.FunctionComponent<IInput> = ({
+    type = "text",
+    className,
+    value,
+
+    attributes = {}
+}) => {
+    return <input
+        value={value}
+        {...attributes}
+        type={type}
+        className={className}
+    />;
+};
+
+export interface ILabelInput {
+    label: string;
+    input: IInput;
+    labelWidth?: number;
+    labelMinWidth?: number;
+    labelMaxWidth?: number;
+    width?: number;
+    minWidth?: number;
+    maxWidth?: number;
+    flexDirection?: "row" | "column";
+}
+
+export const LabelInput: React.FunctionComponent<ILabelInput> = ({
+    label = "",
+    width,
+    minWidth = 60,
+    maxWidth = 240,
+    labelWidth,
+    labelMinWidth = 30,
+    labelMaxWidth = 120,
+    flexDirection = "row",
+    input
+}) => {
+    const i18n = label => label;
+    const inputs = [];
+    if (input.type === "number" || input.type === "range") {
+        inputs.push(<Input {...input}/>);
+        inputs.push(<Input {...input} type="range"/>);
```

Comment:
Why are there two inputs in this scenario?

Reply:
It's a slider with a number input next to it. It's super handy to be able to use a slider to test some values OR set a specific value with the number input.

Reply:
...now that I'm looking at it again it *could* be a range with a range next to it if `type="range"` was passed in.

Reply:
For range yes - but for just numeric input, I would have thought a "spinner" would be the norm?

Reply:
It's super handy to have an editable number input alongside the slider... best of both worlds... sometimes it can be frustrating using a slider and trying to set a specific value.

Reply:
Agree with usability - but that should be a SliderInput.

Also I would make Label an attribute of Input and not the other way round?